src/h5web/visualizations/nexus/utils.ts

```diff
@@ -92,6 +68,50 @@ export function findNxDataGroup(
   return findNxDataGroup(defaultEntity, metadata);
 }
 
+export function findSignalName(group: HDF5Group): string {
+  const signal = getAttributeValue(group, 'signal');
+
+  assertDefined(signal, "Expected 'signal' attribute");
+  assertStr(signal, "Expected 'signal' attribute to be a string");
+
+  return signal;
+}
+
+export function findSignalDataset(
+  signalName: string,
+  group: HDF5Group,
+  metadata: HDF5Metadata
+): HDF5Dataset<HDF5SimpleShape, HDF5NumericType> {
+  const dataset = getLinkedEntity(signalName, group, metadata);
+
+  assertDefined(dataset, `Expected "${signalName}" signal entity to exist`);
+  assertDataset(dataset, `Expected "${signalName}" signal to be a dataset`);
+  assertNumericType(dataset);
+  assertSimpleShape(dataset);
+
+  return dataset;
+}
+
+export function isNxInterpretation(
+  attrValue: HDF5Value
+): attrValue is NxInterpretation {
+  return (
+    typeof attrValue === 'string' &&
+    Object.values<string>(NxInterpretation).includes(attrValue)
+  );
+}
```

Comment:
I've turned this one into a type guard, removed the unnecessary `!attrValue` check and inlined `NX_INTERPRETATIONS`, for consistency with `isScaleType`.